### PR TITLE
Remove manual repro steps for a bash script

### LINKED ISSUE #88: Can we emit server repro package install steps as a script?
Right now, they're in prose because the syntax is different for every command line (bash vs cmd vs powershell).  We could give multiple forms (e.g. under `<summary>` tags, but we have to be careful not to exceed GH's character limit.  This may be less of an issue now that we have [better monorepo support](https://github.com/microsoft/typescript-error-deltas/pull/83).

### PR DESCRIPTION
**Pull request by armanio123**
Fixes #88 

Removes the prose repro steps for a bash script instead. 

It's fairly easy to add other shells but I'm starting with Bash as the project already depends on Linux. Based on feedback I'll decide if it's worth adding them.
 
Note that when running the tsserver user-tests, there's no command for downloading the repository as I was not able to get the information easily. It's still added as comment.

### CODE REVIEW

**Comment by jakebailey** [comment]
I'm not sure that I agree with this, honestly; I know Anders will follow these instructions for doing repros, so putting that behind a bash script will probably slow things down. What is motivating this change?

**Comment by armanio123** [comment]
I have been triaging a couple of this repros and quickly feel like it would be much easier to execute a script. After running a couple, copy-pasting or simply running the steps feel like an annoying task.

Do you know what Anders execute or why he would prefer the steps? I was considering scripting it on powershell or cmd but decided for Bash as I mentioned.

I could add the steps as comments above each command instead or have it on a different section under `Bash script` or alike.

**Comment by jakebailey** [comment]
I think having the script is totally fine, but I would still try and leave those instructions like we do now. Then people who want a quick script can do so.

(But honestly, I usually don't follow the install steps at all; they are usually quite verbose and overkill for getting to the minimal issue)

**Comment by armanio123** [comment]
Do you have a preference on how to include the repro steps? I'm thinking on options: 
1. Leave the prose steps as is and add the bash script after them.
2. Include the prose step on each command of the script

**Comment by jakebailey** [comment]
The first one is what I'd do.

**Comment by armanio123** [comment]
[](url)

File: src/postGithubIssue.ts [hunk 1]
```diff
@@ -72,7 +72,10 @@ ${Object.keys(statusCounts).sort().map(status => `| ${status} | ${statusCounts[s
 `;
 
 const resultPaths = pu.glob(resultDirPath, `**/*.${resultFileNameSuffix}`).sort((a, b) => path.basename(a).localeCompare(path.basename(b)));
-const outputs = resultPaths.map(p => fs.readFileSync(p, { encoding: "utf-8" }).replace(new RegExp(artifactFolderUrlPlaceholder, "g"), artifactsUri));
+const outputs = resultPaths.map(p =>
+    fs.readFileSync(p, { encoding: "utf-8" })
+        .replace(new RegExp(artifactFolderUrlPlaceholder, "g"), artifactsUri)
+        .replace(new RegExp(getArtifactsApiUrlPlaceholder, "g"), getArtifactsApi));
```
Comment (jakebailey) [inline src/postGithubIssue.ts:78]: These could just be replaceAll (I believe we run in node 18+ only)

File: test/__snapshots__/main.test.ts.snap [hunk 2]
```diff
@@ -45,23 +45,45 @@ Raw error text: <code>artifacts/MockRepoOwner.MockRepoName.rawError.txt</code> i
...
+git -C "./MockRepoName" reset --hard 57b462387e88aa7e363af0daf867a5dc1e83a935
+# Install packages (exact steps are below, but it might be easier to follow the repo readme)
+npm --prefix "./dirA" install --prefer-offline --no-audit --no-progress --legacy-peer-deps --ignore-scripts -q
+npm --prefix "./dirB/dirC" install --prefer-offline --no-audit --no-progress --legacy-peer-deps --ignore-scripts -q
+npm --prefix "./dirD/dirE/dirF" install --prefer-offline --no-audit --no-progress --legacy-peer-deps --ignore-scripts -q
+downloadUrl=$(curl -s "PLACEHOLDER_GETARTIFACTS_API?artifactName=RepoResults123&api-version=7.0" | jq -r ".resource.downloadUrl")
+wget -O RepoResults123.zip "$downloadUrl"
+unzip -p RepoResults123.zip RepoResults123/MockRepoOwner.MockRepoName.replay.txt > MockRepoOwner.MockRepoName.replay.txt
+npm install --no-save @typescript/server-replay
+\`\`\`
```
Comment (jakebailey) [inline test/__snapshots__/main.test.ts.snap:72]: Can we just have one of these? Maybe just the latter?

I beleive the original PR had removed the instructions in favor of a download link to a shell script, which is what I was saying would be less usable, but I think this has now jumped to the other side. I think including a shell script block is totally sufficient.

Comment (armanio123) [reply test/__snapshots__/main.test.ts.snap:72]: Sorry for the confusion. This PR introduced the "Bash script" to be shown and replace the "prose steps". I thought you wanted to have both of them.

For clarification, you prefer NO prose steps and ONLY the bash commands as below, correct?

image

Comment (jakebailey) [reply test/__snapshots__/main.test.ts.snap:72]: Yes, super sorry about the confusion...

File: src/postGithubIssue.ts (hunk 1 above)
Comment (armanio123) [reply src/postGithubIssue.ts]: Done.

File: test/__snapshots__/main.test.ts.snap (hunk 2 above)
Comment (armanio123) [reply test/__snapshots__/main.test.ts.snap]: No worries! Done.

**Review by jakebailey — APPROVED** [review]
Thanks!

**Pull request merged**